support-metrics-client/src/main/java/io/confluent/support/metrics/SupportedKafka.java

```diff
@@ -48,8 +45,8 @@ public static void main(String[] args) throws Exception {
       final SupportedServerStartable supportedServerStartable =
           new SupportedServerStartable(serverProps);
 
-      // register signal handler to log termination due to SIGTERM, SIGHUP and SIGINT (control-c)
-      registerLoggingSignalHandler();
+      if (!isWindows() && !isIbmJdk())
```

Comment:
any reason we shouldn't have the signal handler handle those cases directly?

Reply:
That's what I had done first. But because there are many reasons for failure and we want to propagate some of them, I thought it was easier to understand for `register` to throw all exceptions and for the caller to handle them. In the context of AK anyway. We are relying on internal Kafka APIs in CP for now, which is not great. Once there is a replacement for `sun.misc.Signal`, we can simplify it and avoid the CP -> AK dependency.